machineconfiguration/v1/tests/machineconfignodes.machineconfiguration.openshift.io/ImageModeStatusReporting.yaml

```diff
@@ -0,0 +1,188 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "[FeatureGate] ImageModeStatusReporting"
+crdName: machineconfignodes.machineconfiguration.openshift.io
+featureGates:
+- ImageModeStatusReporting
+tests:
+  onCreate:
+    - name: Should be able to create MachineConfigNode when spec.configImage ImageModeStatusReporting is enabled
+      initial: |
+        apiVersion: machineconfiguration.openshift.io/v1
+        kind: MachineConfigNode
+        metadata:
+          name: test-imagemodestatusreporting
+        spec:
+          node:
+            name: test-imagemodestatusreporting
+          pool:
+            name: worker
+          configVersion:
+            desired: rendered-worker-abc
+          configImage: 
+            desiredImage: image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/ocb-image@sha256:7a539f562d8d5d3b6bd7338ac014e34dabc9626cf344d30e412ef98a55cc1bf6
+      expected: |
+        apiVersion: machineconfiguration.openshift.io/v1
+        kind: MachineConfigNode
+        metadata:
+          name: test-imagemodestatusreporting
+        spec:
+          node:
+            name: test-imagemodestatusreporting
+          pool:
+            name: worker
+          configVersion:
+            desired: rendered-worker-abc
+          configImage:
+            desiredImage: image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/ocb-image@sha256:7a539f562d8d5d3b6bd7338ac014e34dabc9626cf344d30e412ef98a55cc1bf6
+    - name: spec.configImage.desiredImage should enforce MaxLength.
+      initial: |
+        apiVersion: machineconfiguration.openshift.io/v1
+        kind: MachineConfigNode
+        metadata:
+          name: test-imagemodestatusreporting
+        spec:
+          node:
+            name: test-imagemodestatusreporting
+          pool:
+            name: worker
+          configVersion:
+            desired: rendered-worker-abc
+          configImage:
+            desiredImage: registry.example.com/a/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very//very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/very/j
+      expectedError: "the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
```

Comment:
Is this the correct error for the data being too long? Were other errors returned at the same time?